eventheader/src/_internal.rs

```diff
@@ -80,3 +85,107 @@ pub const fn time_from_duration_before_1970(duration: Duration) -> i64 {
         -(duration_secs as i64) - ((duration.subsec_nanos() != 0) as i64)
     }
 }
+
+/// Copies the specified value to the specified location.
+/// Returns the pointer after the end of the copy.
+///
+/// # Safety
+///
+/// Caller is responsible for making sure there is sufficient space in the buffer.
+unsafe fn append_bytes<T: Sized>(dst: *mut u8, src: &T) -> *mut u8 {
+    let size = mem::size_of::<T>();
+    ptr::copy_nonoverlapping(src as *const T as *const u8, dst, size);
+    return dst.add(size);
+}
+
+/// Fills in `data[0]` with the event's write_index, event_header,
+/// activity extension block (if an activity id is provided), and
+/// metadata extension block header (if meta_len != 0), then sends
+/// the event to the `user_events_data` file.
+///
+/// Requires:
+/// - `data[0].is_empty()` since it will be used for the headers.
+/// - related_id may only be present if activity_id is present.
+/// - if activity_id.is_some() || meta_len != 0 then event_header.flags
+///   must equal DefaultWithExtension.
+/// - If meta_len != 0 then `data[1]` starts with metadata extension
+///   block data.
+pub fn write_eventheader(
```

Comment:
Note: This used to be a method on `TracepointState`, but that is now in the `tracepoint` crate, and eventheader stuff doesn't belong in that crate, so it is now a free function.